source/MapPanel.cpp

```diff
@@ -899,6 +923,38 @@ void MapPanel::UpdateCache()
 	cachedCommodity = commodity;
 	nodes.clear();
 
+	// Get danger level range so we can scale by it.
+	double dangerMax = 0.;
+	double dangerScale = 1.;
+	if(commodity == SHOW_DANGER)
+	{
+		// Scale danger to span [0, 1] based on known systems, without including raid fleets.
+		double dangerMin = numeric_limits<double>::max();
+		for(const auto &it : GameData::Systems())
+		{
+			const System &system = it.second;
+
+			// Only check displayed systems.
+			if(!system.IsValid() || system.Inaccessible())
+				continue;
+			if(!player.HasSeen(system) && &system != specialSystem)
+				continue;
```

Comment:
This if doesn't seem necessary. The player shouldn't know about the danger level of special systems that haven't been seen, which is caught by the has visited check below.